rust/feature-flags/src/cohort_operations.rs

```diff
@@ -0,0 +1,351 @@
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+use tracing::instrument;
+
+use crate::cohort_models::{Cohort, CohortId, CohortOrEmpty, CohortRow, InnerCohortProperty};
+use crate::{api::FlagError, database::Client as DatabaseClient, flag_definitions::PropertyFilter};
+
+impl Cohort {
+    /// Returns a cohort from postgres given a cohort_id and team_id
+    #[instrument(skip_all)]
+    pub async fn from_pg(
+        client: Arc<dyn DatabaseClient + Send + Sync>,
```

Comment:
Took a crack at this and couldn't get it working nicely, but I want to come back to it – I'll take a gander at it when I revisit error handling, since those two concepts are linked IMO.